### PR TITLE
Overlay events; Adapter service z-index calculations

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
- Added `SkyOverlayConfig.enablePointerEvents`.
- Added `SkyOverlayInstance.backdropClick`.
- Added `SkyCoreAdapterService.isTargetAboveElement()`.
- Fixed bug with `SkyOverlayInstance.closed` subscription.

### CODE REVIEW

File: src/app/public/modules/adapter-service/adapter.service.spec.ts [hunk 1]
```diff
@@ -186,4 +191,80 @@ describe('Core adapter service', () => {
...
+      div2.style.zIndex = '2';
+
+      // Only add the first element.
+      container.appendChild(div1);
+
+      const result = adapter.isTargetAboveElement(div2, div1);
+      expect(result).toEqual(true);
+    });
+
+    it('should walk parents until z-index is found', () => {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/adapter-service/adapter.service.spec.ts]: "should walk"? Not sure what that means here...

File: src/app/public/modules/adapter-service/adapter.service.ts [hunk 2]
```diff
@@ -161,6 +161,36 @@ export class SkyCoreAdapterService {
     return elements;
   }
 
+  /**
+   * Checks if an event target has a higher z-index than a given element.
+   * @param target The event target element.
+   * @param elementRef The element to test against. A z-index must be explicitly set for this element.
+   */
+  public isTargetAboveElement(target: EventTarget, element: HTMLElement): boolean {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/adapter-service/adapter.service.ts]: Would this be more reusable if the `target` argument was an `HTMLElement`?

File: src/app/public/modules/overlay/overlay.component.ts [hunk 3]
```diff
@@ -48,6 +52,8 @@ import {
   SkyOverlayContext
 } from './overlay-context';
 
+let uniqueZIndex = 1001; // Omnibar is 1000
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/overlay/overlay.component.ts]: Should this be called `baselineIndex` or `startingIndex`? I'm not really sure what "unique" means unless I had prior knowledge of the omnibar.

File: src/app/public/modules/adapter-service/adapter.service.spec.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/adapter-service/adapter.service.spec.ts]: "Walking an array" is another way of saying "iterating over an array of ancestors recursively". Python and PHP use the term often. If you hate it, we can change it.

File: src/app/public/modules/adapter-service/adapter.service.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/adapter-service/adapter.service.ts]: I was hoping to communicate with the method parameters that the first "element" should always be the event's "target", not just some element.

File: src/app/public/modules/overlay/overlay.component.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/overlay/overlay.component.ts]: I can rename it, sure. The idea I was trying to communicate was that every overlay would get a unique, auto-incrementing z-index.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/overlay/overlay.component.ts]: Disregard.

File: src/app/public/modules/adapter-service/adapter.service.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/adapter-service/adapter.service.ts]: Got it. Maybe deliberate is better in this situation. I see you also spell that out in the description. I did notice your jsdoc variable is wrong though `elementRef` --> `element`.

**Pull request merged**